cost-analyzer/templates/aggregator-deployment.yaml

```diff
@@ -0,0 +1,193 @@
+              path: /healthz
+              port: 9004
+            initialDelaySeconds: 10
+            periodSeconds: 5
+            failureThreshold: 200
+          imagePullPolicy: Always
+          args: ["waterfowl"]
+          ports:
+            - name: tcp-api
+              containerPort: 9004
+              protocol: TCP
+          resources:
+{{ toYaml .Values.kubecostAggregator.resources | indent 12 }}
+          volumeMounts:
+            - name: persistent-configs
+              mountPath: /var/configs
+            - name: bucket-config
+              mountPath: /var/configs/etl
+            - name: aggregator-storage
+              mountPath: /var/configs/waterfowl
+            {{- if .Values.kubecostAggregator.aggregatorDbStorage }}
+            - name: aggregator-db-storage
+              mountPath: /var/configs/waterfowl/duckdb
+            {{- end }}
+          env:
+            {{- if and (.Values.prometheus.server.global.external_labels.cluster_id) (not .Values.prometheus.server.clusterIDConfigmap) }}
+            - name: CLUSTER_ID
+              value: {{ .Values.prometheus.server.global.external_labels.cluster_id }}
+            {{- end }}
+            {{- if .Values.prometheus.server.clusterIDConfigmap }}
+            - name: CLUSTER_ID
+              valueFrom:
+                configMapKeyRef:
+                  name: {{ .Values.prometheus.server.clusterIDConfigmap }}
+                  key: CLUSTER_ID
+            {{- end }}
+
+            - name: CONFIG_PATH
+              value: /var/configs/
+            - name: ETL_ENABLED
+              value: "false" # this pod should never run KC's concept of "ETL"
+            - name: CLOUD_PROVIDER_API_KEY
+              value: "AIzaSyDXQPG_MHUEy9neR7stolq6l0ujXmjJlvk" # The GCP Pricing API key.This GCP api key is expected to be here and is limited to accessing google's billing API.'
+              value: "true" # just in case, not sure if necessary
+
+            {{- if $etlBackupBucketSecret }}
+            # If this isn't set, we pretty much have to be in a read only state,
+            # initialization will probably fail otherwise.
+            - name: ETL_BUCKET_CONFIG
+              {{- if not .Values.kubecostModel.federatedStorageConfigSecret}}
+              value: "/var/configs/etl/object-store.yaml"
+              {{- else  }}
+              value: "/var/configs/etl/federated-store.yaml"
+            - name: FEDERATED_STORE_CONFIG
+              value: "/var/configs/etl/federated-store.yaml"
+            - name: FEDERATED_PRIMARY_CLUSTER # this ensures the ingester runs assuming federated primary paths in the bucket
+              value: "true"
+            - name: FEDERATED_CLUSTER # this ensures the ingester runs assuming federated primary paths in the bucket
+              value: "true"
+              {{- end }}
+            {{- end }}
+
+            # TODO (michaelmdresser):
+            # - SAML configs?
+            # - Product key configs?
```

Comment:
I'll file Jira tickets to look into these before 1.107.

Reply:
removed these comments since you are tracking in jira now